akd_local_auditor/src/auditor/mod.rs

```diff
@@ -0,0 +1,215 @@
+    result.extend(header);
+    result.extend(epoch_bytes);
+    result.extend(p_bytes);
+    result.extend(c_bytes);
+    result
+}
+
+pub async fn audit_epoch<H>(blob: akd::proto::AuditBlob, qr: bool) -> Result<()>
+where
+    H: winter_crypto::Hasher + Sync + Send,
+{
+    let (p_hash, c_hash, epoch, proof) = blob.decode().map_err(|err| anyhow!("{}", err))?;
+
+    // DO THE FREAKING VERIFICATION ALREADY
+    if let Err(akd_error) = akd::auditor::audit_verify(
+        vec![p_hash, c_hash],
+        akd::proof_structs::AppendOnlyProof::<crate::Hasher> {
+            proofs: vec![proof],
+            epochs: vec![epoch],
+        },
+    )
+    .await
+    {
+        warn!(
+            "Audit proof for epoch {} failed to verify with error {}",
+            epoch, akd_error
+        );
+    } else {
+        info!("Audit proof for epoch {} has verified!", epoch);
+        if qr {
+            info!("Generating scan-able QR code for the verification on device");
+            let qr_code_data = format_qr_record::<crate::Hasher>(p_hash, c_hash, epoch);
+            if let Err(error) = qr2term::print_qr(&qr_code_data) {
+                error!("Error generating QR code {}", error);
+                bail!("Error generating QR code {}", error);
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Storage options for retrieving audit proofs
+#[derive(Subcommand, Clone, Debug)]
+pub enum AuditCommand {
+    /// Audit a specific epoch
+    Audit {
+        /// The epoch to audit
+        #[clap(long)]
+        epoch: u64,
+        /// Show a QR code of the audit
+        #[clap(long)]
+        qr: bool,
+    },
+    /// Show the available epochs to audit
+    ShowEpochs,
+}
+
+/// Audit opertions supported by the client
+#[derive(Parser, Clone, Debug)]
+pub struct AuditArgs {
+    #[clap(subcommand)]
+    command: AuditCommand,
+}
+
+fn display_audit_proofs_info(info: &mut Vec<crate::storage::EpochSummary>) -> Result<()> {
+    info.sort_by(|a, b| a.epoch.cmp(&b.epoch));
+    if info.is_empty() {
+        bail!("There are no epochs present in the storage repository");
+    }
+
+    let min = info.first().unwrap().clone();
+    let max = info.last().unwrap().clone();
+    let (maybe_broken_epoch, is_contiguous) =
+        info.iter()
+            .skip(1)
+            .fold((min.clone(), true), |(previous_item, cont), item| {
+                if !cont {
+                    (previous_item, cont)
+                } else {
+                    (item.clone(), item.epoch == previous_item.epoch + 1)
+                }
+            });
+
+    if !is_contiguous {
+        bail!("The audit proofs appear to not be continguous. There's a break in the linear history at epoch {}", maybe_broken_epoch.epoch);
+    }
+
+    info!(
+        "Audit history is available between epochs ({}) and ({}), inclusively.",
+        min.epoch, max.epoch
+    );
+
+    Ok(())
+}
+
+async fn process_command(
+    storage: &Box<dyn crate::storage::AuditProofStorage>,
```

Comment:
As the linter suggests, is the `Box` really necessary here?

Reply:
Once again the nightly clippy strikes! Local clippy shows nothing :p